### PR TITLE
vapoursynth-bestsource: init at R6

### PR DESCRIPTION
**Pull request by snaakey**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/vapoursynth-bestsource/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  meson,
+  ninja,
+  cmake,
+  pkg-config,
+  vapoursynth,
+  ffmpeg,
+  xxHash,
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Sigmanificient) [inline pkgs/development/libraries/vapoursynth-bestsource/default.nix:14]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/development/libraries/vapoursynth-bestsource/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    description = "Wrapper library around FFmpeg that ensures sample and frame accurate access to audio and video";
+    homepage = "https://github.com/vapoursynth/bestsource";
+    license = with licenses; [
+      mit
+      wtfpl
+      gpl2Plus
+    ];
+    maintainers = with maintainers; [ snaki ];
+  };
+}
```
Comment (Sigmanificient) [inline pkgs/development/libraries/vapoursynth-bestsource/default.nix:56]:
```suggestion
})
```

File: pkgs/development/libraries/vapoursynth-bestsource/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  meta = with lib; {
+    description = "Wrapper library around FFmpeg that ensures sample and frame accurate access to audio and video";
+    homepage = "https://github.com/vapoursynth/bestsource";
+    license = with licenses; [
+      mit
+      wtfpl
+      gpl2Plus
+    ];
+    maintainers = with maintainers; [ snaki ];
+  };
```
Comment (Sigmanificient) [inline pkgs/development/libraries/vapoursynth-bestsource/default.nix:55]: - #208242
```suggestion
  meta = {
    description = "Wrapper library around FFmpeg that ensures sample and frame accurate access to audio and video";
    homepage = "https://github.com/vapoursynth/bestsource";
    license = with lib.licenses; [
      mit
      wtfpl
      gpl2Plus
    ];
    maintainers = with lib.maintainers; [ snaki ];
  };
```

File: pkgs/development/libraries/vapoursynth-bestsource/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    fetchSubmodules = true;
+    owner = "vapoursynth";
+    repo = "bestsource";
+    rev = version;
```
Comment (Sigmanificient) [inline pkgs/development/libraries/vapoursynth-bestsource/default.nix:27]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by Sigmanificient** [comment]
Can you please define your package within `pkgs/by-name/va/vapoursynth-bestsource/package.nix`?

File: pkgs/by-name/va/vapoursynth-bestsource/package.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    fetchSubmodules = true;
+    owner = "vapoursynth";
+    repo = "bestsource";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-ICkdIomlkHUdK6kMeui45fvUn4OMxSrP8svB2IN+GCg=";
+  };
```
Comment (Sigmanificient) [inline pkgs/by-name/va/vapoursynth-bestsource/package.nix:29]: Oh, the `R` shouldn't be feature in the derivation, only within the tag. Not sure if version should be set to `6` or `6.0.0` though

```suggestion
  version = "6";

  outputs = [
    "out"
    "dev"
  ];

  src = fetchFromGitHub {
    fetchSubmodules = true;
    owner = "vapoursynth";
    repo = "bestsource";
    rev = "refs/tags/R${finalAttrs.version}";
    hash = "sha256-ICkdIomlkHUdK6kMeui45fvUn4OMxSrP8svB2IN+GCg=";
  };
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by illdefined** [comment]
Result of `nixpkgs-review pr 331874` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vapoursynth-bestsource</li>
    <li>vapoursynth-bestsource.dev</li>
  </ul>
</details>

**Review by Sigmanificient — APPROVED** [review]

**Review by illdefined — APPROVED** [review]

**Comment by illdefined** [comment]
Result of `nixpkgs-review pr 331874` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vapoursynth-bestsource</li>
    <li>vapoursynth-bestsource.dev</li>
  </ul>
</details>

File: pkgs/by-name/va/vapoursynth-bestsource/package.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  buildInputs = [
+    vapoursynth
+    ffmpeg
+    xxHash
+  ];
+
+  postPatch = ''
+    substituteInPlace meson.build \
+      --replace-fail "vapoursynth_dep.get_variable(pkgconfig: 'libdir')" "get_option('libdir')"
```
Comment (illdefined) [inline pkgs/by-name/va/vapoursynth-bestsource/package.nix:46]: I personally like this approach a lot better than a patch.

**Review by illdefined — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/va/vapoursynth-bestsource/package.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  '';
+
+  meta = {
```
Comment (leona-ya) [inline pkgs/by-name/va/vapoursynth-bestsource/package.nix:49]: This should have a `platforms` attribute limited to x86_64, as this is (according to ofborg) the only supported platform for vapoursynth.

**Review by leona-ya — APPROVED** [review]

**Pull request merged**